example/zipIterator/src/zipIterator-main.cpp

```diff
@@ -189,45 +201,45 @@ int main()
     
     std::cout << "zipIter[9]: ";
     printTuple(zipIter[9]);
-    std::cout << "\n\n";
-
-
-
-    std::cout << "-----\n\n"
-              << "Failed when building for accelerator testing\n\n";
-
-    // std::cout << "Testing accelerator: " << alpaka::getAccName<TAcc>() << " with size: " << n << "\n\n";
-
-    // IteratorTuple deviceZipTuple = std::make_tuple(deviceNative, deviceNativeChar, deviceNativeDouble);
-    // vikunja::mem::iterator::ZipIterator<IteratorTuple> deviceZipIter(deviceZipTuple);
-
-    // using IteratorTupleReduceResult = std::tuple<TRed, TRedChar, TRedDouble>;
-
-    // // Use Lambda function for reduction
-    // auto sum = [] ALPAKA_FN_HOST_ACC(IteratorTuple const i, IteratorTuple const j)
-    // {
-    //     IteratorTupleReduceResult tmp = std::make_tuple(*std::get<0>(i) + *std::get<0>(j), *std::get<1>(i), *std::get<2>(i) + *std::get<2>(j));
-    //     return tmp;
-    // };
-    // auto doubleNum = [] ALPAKA_FN_HOST_ACC(IteratorTuple const i)
-    // {
-    //     IteratorTupleReduceResult tmp = std::make_tuple(2 * *std::get<0>(i), *std::get<1>(i), 2 * *std::get<2>(i));
-    //     return tmp;
-    // };
-
-    // // TRANSFORM_REDUCE CALL:
-    // // Takes the arguments: accelerator device, host device, accelerator queue, size of data, pointer-like to memory,
-    // // transform lambda, reduce lambda.
-    // auto transformReduceResult = vikunja::reduce::deviceTransformReduce<TAcc>(
-    //     devAcc,
-    //     devHost,
-    //     queueAcc,
-    //     n,
-    //     deviceZipIter,
-    //     doubleNum,
-    //     sum);
-
-
+    std::cout << "\n\n"
+              << "-----\n\n";
+
+    // Revert the number values for index 4
+    zipIter = std::make_tuple(std::get<0>(*zipIter) / 2, std::get<1>(*zipIter), std::get<2>(*zipIter) / 2);
+
+    IteratorTuplePtr deviceZipTuple = std::make_tuple(deviceNative, deviceNativeChar, deviceNativeDouble);
+    vikunja::mem::iterator::ZipIterator<IteratorTuplePtr, IteratorTupleVal> deviceZipIter(deviceZipTuple);
+
+    auto doubleNum = [] ALPAKA_FN_HOST_ACC(IteratorTupleVal const i)
+    {
+        return std::make_tuple(2 * std::get<0>(i), std::get<1>(i), 2 * std::get<2>(i));
+    };
+
+    vikunja::transform::deviceTransform<TAcc>(
+        devAcc,
+        queueAcc,
+        extent[Dim::value - 1u],
+        deviceZipIter,
+        deviceZipIter,
```

Comment:
That's wrong. You cannot write your output to a fancy iterator. It needs to be a memory object.